sklearn/naive_bayes.py

```diff
@@ -951,3 +964,182 @@ def _joint_log_likelihood(self, X):
+    discrete features, that are categorically distributed. Each feature is sampled
+    by its own categorical distribution.
+
+    Read more in the :ref:`User Guide <categorical_naive_bayes>`.
+
+    Parameters
+    ----------
+    alpha : float, optional (default=1.0)
+        Additive (Laplace/Lidstone) smoothing parameter
+        (0 for no smoothing).
+
+    fit_prior : boolean, optional (default=True)
+        Whether to learn class prior probabilities or not.
+        If false, a uniform prior will be used.
+
+    class_prior : array-like, size (n_classes,), optional (default=None)
+        Prior probabilities of the classes. If specified the priors are not
+        adjusted according to the data.
+
+    Attributes
+    ----------
+    class_log_prior_ : array, shape (n_classes, )
+        Smoothed empirical log probability for each class.
+
+    feature_log_prob_ : array, shape (n_classes, n_features)
+        Empirical log probability of features
+        given a class, ``P(x_i|y)``.
+
+    class_count_ : array, shape (n_classes,)
+        Number of samples encountered for each class during fitting. This
+        value is weighted by the sample weight when provided.
+
+    feature_count_ : dict of arrays, len n_features
+        Holds an array of shape () for each feature. Each array provides
+        the number of samples encountered for each class and category
+        of the specific feature.
+
+    n_features_ : int
+        Number of features of each sample.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> X = np.random.randint(5, size=(6, 100))
+    >>> y = np.array([1, 2, 3, 4, 5, 6])
+    >>> from sklearn.naive_bayes import CategoricalNB
+    >>> clf = CategoricalNB()
+    >>> clf.fit(X, y)
+    CategoricalNB(alpha=1.0, class_prior=None, fit_prior=True)
+    >>> print(clf.predict(X[2:3]))
+    [3]
+    """
+
+    def __init__(self, alpha=1.0, fit_prior=True, class_prior=None):
+        if np.__version__ < '1.9.0':
+            warnings.warn(('numpy is older than 1.9.0. Therefore assure that X is castable to int without loss of '
+                           'information and all features of X are greater or equal 0.'))
+        self.alpha = alpha
+        self.fit_prior = fit_prior
+        self.class_prior = class_prior
+
+    def fit(self, X, y):
+        """Fit Naive Bayes classifier according to X, y
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features. Here, each feature of X is
+            assumed to be from a different categorical distribution.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        Returns
+        -------
+        self : object
+        """
+        if not np.__version__ >= '1.9.0':
+            check_array(X, accept_large_sparse=False)
```

Comment:
can you use an int dtype here?

Reply:
No, I tried it, but I can't. If I pass `dtype=np.int64` as argument to `check_array`, `np.inf`'s will be converted to integers. I tried to make some modifications to the code, but it then breaks at other places (regressor bagging). So I would suggest to use the not really pretty, but convenient workaround [here](https://github.com/scikit-learn/scikit-learn/pull/12569/files#diff-dff46bc51c084c731bcbe4cb9b6aa243R1127).
Soon legacy numpy should be obsolete anyways with #12639.